src/Widgets/TerminalWidget.vala

```diff
@@ -331,6 +336,7 @@ namespace Terminal {
         void on_child_exited () {
             child_has_exited = true;
             last_key_was_return = true;
+            check_cwd_changed ();
```

Comment:
why do you have to check here? To my understanding a child process shouldn't be able to change the cwd of the parent.

Reply:
OK, I wasnt sure so I played safe.  Removing.